File: test_db.py
import firebase_admin
from firebase_admin import credentials
from firebase_admin import firestore

try:
    cred = credentials.Certificate('/Users/shaliqrahman/Documents/ZeeSha/SOOCHER COLLAB/soocher-web/soocher-web-booking/service_account.json')
    firebase_admin.initialize_app(cred)
    db = firestore.client()

    doctor_id = 'mtlUtszDdJMYof7e6pRyPkSJrb12'
    slots_ref = db.collection('Users').document(doctor_id).collection('Available Slots')
    docs = slots_ref.stream()
    
    found = False
    for doc in docs:
        found = True
        print(f"{doc.id} => {doc.to_dict()}")
    
    if not found:
        print("No documents found in Available Slots")
        # Let's try to list subcollections
        collections = db.collection('Users').document(doctor_id).collections()
        for col in collections:
            print(f"Found subcollection: {col.id}")

except Exception as e:
    print(f"Error: {e}")

File: test_db.js
const admin = require('firebase-admin');
const serviceAccount = require('./service_account.json');

admin.initializeApp({
    credential: admin.credential.cert(serviceAccount)
});

const db = admin.firestore();
const doctorId = 'mtlUtszDdJMYof7e6pRyPkSJrb12';

async function test() {
    const docSnap = await db.collection('Users').doc(doctorId).get();
    if (docSnap.exists) {
        console.log("Doctor exists:", docSnap.data().name, docSnap.data().specialization);
    } else {
        console.log("Doctor does not exist");
    }

    const slotsRef = db.collection('Users').doc(doctorId).collection('Available Slots');
    const snap = await slotsRef.get();

    if (snap.empty) {
        console.log("No documents found in 'Available Slots'");
        const collections = await db.collection('Users').doc(doctorId).listCollections();
        for (const col of collections) {
            console.log('Found subcollection:', col.id);
        }
    } else {
        snap.forEach(doc => {
            console.log(doc.id, '=>', JSON.stringify(doc.data(), null, 2));
        });
    }
}

test().catch(console.error);
